packages/openapi/src/openapi-request-builder.ts

```diff
@@ -164,23 +164,26 @@ export class OpenApiRequestBuilder<ResponseT = any> {
 
   private getPath(): string {
     const pathParameters = this.parameters?.pathParameters || {};
+    let path = this.pathPattern;
 
-    const pathParts = this.pathPattern.split('/');
-    return pathParts
-      .map(part => {
-        if (OpenApiRequestBuilder.isPlaceholder(part)) {
-          const paramName = part.slice(1, -1);
-          const paramValue = pathParameters[paramName];
-          if (!paramValue) {
-            throw new Error(
-              `Cannot execute request, no path parameter provided for '${paramName}'.`
-            );
-          }
-          return encodeURIComponent(paramValue);
-        }
-        return part;
-      })
-      .join('/');
+    // Get the innermost curly bracket pairs with non-empty and legal content as placeholders.
+    const placeholders = path.match(/{[^/?#{}]+}/g);
+    if (!placeholders) {
+      return path;
+    }
+
+    placeholders.map(placeholder => {
```

Comment:
If you don't return anything, you don't need to map 😊 
```suggestion
    placeholders.forEach(placeholder => {
```

Reply:
You could also use reduce, then you wouldn't have to make "path" modifiable.
```ts
return placeholders.reduce((path, placeholder) => {
  return path.replace(placeholder, encodeURIComponent(parameterValue));
}, this.pathPattern);
```

Reply:
Wonderful solution with `reduce()`! Thanks!